astroid/nodes/node_classes.py

```diff
@@ -2697,6 +2696,37 @@ def _infer_name(self, frame, name):
         return name
 
 
+class ParamSpec(_base_nodes.AssignTypeNode):
+    """Class representing a :class:`ast.ParamSpec` node.
+
+    >>> import astroid
+    >>> node = astroid.extract_node('type Alias[**P] = Callable[P, int]')
+    >>> node.type_params[0]
+    <ParamSpec l.1 at 0x7f23b2e4e198>
+    """
+
+    def __init__(
+        self,
+        lineno: int | None = None,
```

Comment:
Are any of these ever actually unknown? Now that we are 3.8+ they should always be not-None right?

Reply:
I don't know the backstory on this one, I just copy-pasted. It's the same copy-paste operation later when fixing everywhere once we've investigated.

Reply:
I think it makes sense to make them required now. We're doing similar things for the other constructors. I haven't found a way to make it clear to `mypy` but it would be nice for it to know that certain nodes always have a `lineno`.